File: Python/Best Time to Buy and Sell Stock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        if prices == []:
            return 0
        max_profit = 0
        buy_price = prices[0]
        for i in prices:
            max_profit = max(max_profit, i - buy_price)
            buy_price = min(buy_price, i)
        return max_profit
        

File: cpp/Best Time to Buy and Sell Stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        if (prices.size() == 0)
            return 0;
        int max_profit = 0;
        int buy_price = prices[0];
        for (int price : prices) {
            max_profit = max(max_profit, price - buy_price);
            buy_price = min(buy_price, price);
        }
        return max_profit;
    }
};
